sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -665,6 +665,9 @@ private Map<String, Object> overrideBootstrapServersConfig(
     return config;
   }
 
+  // TODO: Collapse the two moving average trackers into a single accumulator using a single Guava
+  // AtomicDouble. Note that this requires that a single thread will call update and that while get
+  // may be called by multiple threads the method must only load the accumulator itself.
```

Comment:
I think my confusion is due to getTotalSize taking param named numRecords. It seems to instead be the partition backlog bytes and then the math makes more sense.

Can we rename it and rename it at the call-site where it is also numRecords?

Reply:
Renamed `numRecords` to `estimatedNumRecords` at the call-site and removed the method's parameter after trying to keep it as either `maxNumRecords` or `estimatedNumRecords`. Instead, the method `getGapsFilledWithZeros` omits scaling by any parameter and leaves that to the caller. This allows the caller to provide an appropriate name for a potential scaling factor at the call-site (`currentNumRecords`, `maxNumRecords`, `estimatedNumRecords`, `expectedNumRecords`, etc.).